src/TurbulenceClosures/vertically_implicit_diffusion_solver.jl

```diff
@@ -0,0 +1,185 @@
+using Oceananigans.Operators: interpolation_code
+using Oceananigans.AbstractOperations: flip
+using Oceananigans.Solvers: BatchedTridiagonalSolver, solve!
+
+#####
+##### Vertically implicit solver
+#####
+
+struct VerticallyImplicitDiffusionSolver{A, H, Z}
+    architecture :: A
+    z_center_solver :: H # also used for horizontal velocities
+    z_face_solver :: Z
+end
+
+"""
+    z_viscosity(closure, diffusivities)
+
+Returns the "vertical" (z-direction) viscosity associated with `closure`.
+"""
+function z_viscosity end
+
+"""
+    z_diffusivity(closure, diffusivities, ::Val{tracer_index}) where tracer_index
+
+Returns the "vertical" (z-direction) diffusivity associated with `closure` and `tracer_index`.
+"""
+function z_diffusivity end
+
+implicit_step!(field, ::Nothing, args...; kwargs...) = NoneEvent()
+implicit_diffusion_solver(::ExplicitTimeDiscretization, args...; kwargs...) = nothing
+
+#####
+##### Solver kernel functions for tracers / horizontal velocities and for vertical velocities
+##### Note: "ivd" stands for implicit vertical diffusion.
+#####
+
+@inline κ_Δz²(i, j, kᶠ, kᶜ, grid, κ) = κ / Δzᵃᵃᶜ(i, j, kᶜ, grid) / Δzᵃᵃᶠ(i, j, kᶠ, grid)
+
+# Tracers and horizontal velocities at cell centers in z
+
+@inline function ivd_upper_diagonalᵃᵃᶜ(i, j, k, grid, clock, Δt, κ⁻⁻ᶠ, κ)
+    κᵏ⁺¹ = κ⁻⁻ᶠ(i, j, k+1, grid, clock, κ)
```

Comment:
I assume `⁻⁻ᶠ` is the same as `ᵃᵃᶠ`? If so should we start using it from now on? I'm happy to make the switch.

Reply:
Ah its just notation. I was thinking that `ᵃ` means "Any", as in (any location works here). But here `⁻` is not actually "Any" --- the horizontal locations _do_ matter --- it's just that they're determined by `κ⁻⁻ᶠ` (which is input to the function as an argument) rather than prescribe here in the code. Maybe I can write this more clearly?